src/mod/internal/login_mod.cpp

```diff
@@ -27,6 +27,31 @@
 #include "core/RDP/slowpath.hpp"
 #include "RAIL/client_execute.hpp"
 
+#include <string>
+#include <tuple>
+
+namespace
+{
+    inline std::tuple<std::string, std::string> rpartition(std::string text,
```

Comment:
- `std::string_view` parameter is preferable to `std::string`.
- `std::pair` is preferable to `std::tuple`, save our compilation time :D